### PR TITLE
Support Github App authentication

### PR DESCRIPTION
**Pull request by julienduchesne**
Using a Github App has the benefit that API calls count towards that app installation instead of a user
This is useful in cases where you want to limit the number of automation users but have no such limits on Apps, since they are free

Let me know what you think!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,9 @@ Use the following environment variables to configure the application:
...
 | <tt>GITHUB_UPLOAD_URL</tt>                    | `https://uploads.github.com/` | GitHub upload URL for uploading files. <br> <br>It is taken into account only when `GITHUB_BASE_URL` is also set. If only `GITHUB_BASE_URL` is provided, this parameter defaults to the `GITHUB_BASE_URL` value.                                                                                                                                                                                                                                                |
+| <tt>GITHUB_APP_ID</tt>                        |                               | Github App ID for authentication. This replaces the `GITHUB_ACCESS_TOKEN`.                                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (mszostok) [inline README.md:126]: Could you also create (short or even raw note) on how to use it with it? Similar to the token access doc: https://github.com/mszostok/codeowners-validator/blob/main/docs/gh-token.md

---
How I tested that:
1. Created a GitHub App (https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app).
2. Added the read-only access to members in org:
    Screen Shot 2022-04-14 at 15 03 45

4. Installed the App in my org.

**Obtaining the installation ID:**
1. With a given JS snippet https://github.com/octokit/auth-app.js/#authenticate-as-github-app-json-web-token, I generated a JWT.
2. With the JWT, I listed installations:
    ```bash
    curl -i -X GET \
    -H "Authorization: Bearer <token>" \
    -H "Accept: application/vnd.github.v3+json" \
    https://api.github.com/app/installations
    ```

So finally I had the `GITHUB_APP_ID` and `GITHUB_APP_PRIVATE_KEY ` from GitHub APP setting, and the `GITHUB_APP_INSTALLATION_ID` based on the executed curl. 

Is there an easier way?

**Review by mszostok — COMMENTED** [review]
Hi! 

In general I like the idea, I tested that and works well 👍 

Before merging, please address the `validation` approach. The usage docs are just nice-to-have, and if you don't have time I will add them later 👍

File: README.md (hunk 1 above)
Comment (julienduchesne) [reply README.md]: There is an easier way that doesn't require that JWT stuff (which is quite complex to do if you don't do it often IMO). I've documented it, let me know what you think!

File: internal/github/client.go [hunk 2]
```diff
@@ -2,41 +2,74 @@ package github
...
 	"github.com/mszostok/codeowners-validator/pkg/url"
 
-	"github.com/google/go-github/v29/github"
+	"github.com/google/go-github/v41/github"
 	"golang.org/x/oauth2"
 )
 
 type ClientConfig struct {
-	AccessToken        string
+	AccessToken string `envconfig:"optional"`
```
[the remark this answers is not in the document]
Comment (julienduchesne) [reply internal/github/client.go:19]: Could you run the integration tests update? It seems I can't do it myself

**Comment by julienduchesne** [comment]
Thanks for the great review @mszostok!

File: README.md (hunk 1 above)
Comment (mszostok) [reply README.md]: Looks great! Thanks!

File: internal/github/client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mszostok) [reply internal/github/client.go:18]: No problem, I adjusted golden file and also added new test to ensure no regression when GitHub App opt is used (for example, if I will add new query or sth different I will find out that new perms are needed)

**Comment by mszostok** [comment]
I pushed a dedicated commit https://github.com/mszostok/codeowners-validator/pull/146/commits/1915c98e1686517414ea94d3f0cb5eeac1336763 with e2e test changes. Changes that you applied LGTM, CI is green, so I'm merging the PR. 

Once again, thanks for your contribution! The new release will be done once I will merge the https://github.com/mszostok/codeowners-validator/pull/149 and https://github.com/mszostok/codeowners-validator/pull/148

**Pull request merged**